### PR TITLE
load_shed: make constructor for Overloaded error public

### PR DESCRIPTION
**Pull request by mattklein123**
This allows for mocking. This also matches what is done for
the timeout Elapsed error.

Signed-off-by: Matt Klein <mklein@lyft.com>

### CODE REVIEW

**Comment by mattklein123** [comment]
Any chance of getting this reviewed/merged? Thank you!

**Review by LucioFranco — APPROVED** [review]
Sorry for the delay on this! I was out on vacation :D 

Generally, we don't want to expose these constructors but we already do this for the `Elapsed` error in the timeout layer so I am fine with merging this.

File: tower/src/load_shed/error.rs [hunk 1]
```diff
@@ -12,7 +12,8 @@ pub struct Overloaded {
 }
 
 impl Overloaded {
-    pub(crate) fn new() -> Self {
+    /// Construct a new overloaded error
+    pub fn new() -> Self {
```
Comment (hawkw) [inline tower/src/load_shed/error.rs:16]: tiny style nit: if we're going to have a public 0-argument `new`, we may as well also implement `Default`...

**Review by hawkw — APPROVED** [review]
This seems fine to me.

In general, we try to avoid exposing error type constructors when they're not necessary, as it poses a forward-compatibilty issue (we may want to add additional data to the error in the future), but in this case, it seems pretty unlikely, and if it's useful for testing, we may as well make it public.

File: tower/src/load_shed/error.rs (hunk 1 above)
Comment (mattklein123) [reply tower/src/load_shed/error.rs:17]: Updated, thanks for reviewing.

**Comment by mattklein123** [comment]
Awesome thank you for the merge!

**Pull request merged**